cpp/src/arrow/python/serialize.h

```diff
@@ -107,6 +107,28 @@ Status WriteNdarrayHeader(std::shared_ptr<DataType> dtype,
                           const std::vector<int64_t>& shape, int64_t tensor_num_bytes,
                           io::OutputStream* dst);
 
+struct PythonType {
+  enum type {
+    BOOL,
+    INT,
+    PY2INT,
+    BYTES,
+    STRING,
+    HALF_FLOAT,
+    FLOAT,
+    DOUBLE,
+    DATE64,
+    LIST,
+    DICT,
+    TUPLE,
+    SET,
+    TENSOR,
+    NDARRAY,
+    BUFFER,
+    MAX
```

Comment:
Not sure the name `MAX` is a good idea, it might be a preprocessor-defined macro on some systems...

Reply:
renamed it, good catch